src/library/scala/Product1.scala

```diff
@@ -35,7 +35,7 @@ trait Product1[@specialized(Int, Long, Double) +T1] extends Any with Product {
   @throws(classOf[IndexOutOfBoundsException])
   override def productElement(n: Int) = n match { 
     case 0 => _1
-    case _ => throw new IndexOutOfBoundsException(n.toString())
+    case _ => throw new IndexOutOfBoundsException(s"No element at $n since bound is $productArity")
```

Comment:
This is generated code

Reply:
Updated `genprod`
https://github.com/scala/scala/pull/7210/files#diff-3c847a83a2edf9796c71ee0eb6118d54L363